### PR TITLE
Fix for scaling decimal values in pilight binding

### PR DESCRIPTION
**Pull request by idserda**
Number values received from pilight would have the decimal point in the wrong position for some devices. This fixes that and adds a testcase. It also makes sure the initial state is fetched.

### CODE REVIEW

File: bundles/binding/org.openhab.binding.pilight.test/src/test/java/org/openhab/binding/pilight/internal/PilightGenericBindingProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+		DecimalType number0 = (DecimalType) binding.getState("23", config);
+		Assert.assertEquals(number0.toBigDecimal().compareTo(new BigDecimal("23")), 0);
+
+		config.setScale(1);
+		DecimalType number1 = (DecimalType) binding.getState("23.3", config);
+		Assert.assertEquals(number1.toBigDecimal().compareTo(new BigDecimal("23.3")), 0);
+
+		config.setScale(0);
+		DecimalType number2 = (DecimalType) binding.getState("2.3", config);
+		Assert.assertEquals(number2.toBigDecimal().compareTo(new BigDecimal("23")), 0);
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.pilight.test/src/test/java/org/openhab/binding/pilight/internal/PilightGenericBindingProviderTest.java]: i am not sure if you got your test correctly. But 2.3 should return "2" at scale "0", shouldn't it?

File: bundles/binding/org.openhab.binding.pilight/src/main/java/org/openhab/binding/pilight/internal/PilightBinding.java [hunk 2]
```diff
@@ -105,21 +105,26 @@ private void processValueEvent(List<PilightBindingConfig> configs, Status status
 			String property = config.getProperty();
 			if (status.getValues().containsKey(property)) {
 				String value = status.getValues().get(property);
-				State state = getStateFromProperty(value, config);
-				eventPublisher.postUpdate(config.getItemName(), state);
+				State state = getState(value, config);
+				
+				if (state != null)
+					eventPublisher.postUpdate(config.getItemName(), state);
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.pilight/src/main/java/org/openhab/binding/pilight/internal/PilightBinding.java]: please add curly braces here (even though it's a One-liner)

**Comment by idserda** [comment]
Thanks @teichsta for the review, I've made a few small changes.

**Comment by teichsta** [comment]
>  I've made a few small changes.

Thanks @idserda!

**Pull request merged**